tests/src/Unit/ServerServiceTest.php

```diff
@@ -0,0 +1,148 @@
+        'features' => SOAP_SINGLE_ELEMENT_ARRAYS,
+        'cache_wsdl' => WSDL_CACHE_NONE,
+    );
+
+    /**
+     * Test handling of SoapServer configuration.
+     */
+    public function testSoapConfig()
+    {
+        $clientConfig = new Config(array(
+                'inputFile' => $this->wsdl,
+                'outputDir' => null,
+                'namespaceName' => $this->namespace,
+                'soapClientClass' => $this->soapclientClass,
+                'soapClientOptions' => $this->soapclientOptions,
+            ));
+
+        $clientService = new Service($clientConfig, 'TestClientService', array(), 'Service description');
+        $this->generateClass($clientService, $this->namespace);
+
+        $this->assertClassExists('TestClientService', $this->namespace);
+
+        $serverConfig = new Config(array(
+                'inputFile' => $this->wsdl,
+                'outputDir' => null,
+                'namespaceName' => $this->namespace,
+                'serverClassName' => 'TestAbstractServerService',
+                'soapServerClass' => $this->soapserverClass,
+                'soapServerOptions' => $this->soapserverOptions,
+            ));
+
+        $serverService = new ServerService($serverConfig, $clientService->getClass()->getIdentifier(), array(), 'Service description');
+        $this->generateServerServiceClass($serverService, $this->namespace);
+        // PHP-VCR wants us to define the class VCR\Util\SoapClientTest\TestAbstractServerService
+        // before creating \SoapClientTest\TestAbstractServerService.
+        $this->generateServerServiceClass($serverService, "VCR\\Util\\{$this->namespace}", "\\{$this->namespace}\\TestClientService as TestClientService");
+
+        $this->assertClassExists('TestAbstractServerService', $this->namespace);
+
+        eval("namespace {$this->namespace}; class TestServerService extends \\{$this->namespace}\\TestAbstractServerService {}");
+
+        $serverService = new \SoapClientTest\TestServerService();
+
+        return $serverService;
+    }
+
+    /**
+     * Generate and load a server service class into PHP memory.
+     *
+     * This will cause the server service class to be available for subsequent code.
+     *
+     * @param ClassGenerator $generator The object from which to generate the class.
+     * @param string $namespace The namespace to use for the class.
+     * @param string $useStatement The "use statement" to use for the class.
+     */
+    private function generateServerServiceClass(ClassGenerator $generator, $namespace = null, $useStatement = null)
+    {
+        $source = $generator->getClass()->getSource();
+        if (!empty($useStatement)) {
+            $source = 'use ' . $useStatement . ';' . PHP_EOL . $source;
+        }
+        if (!empty($namespace)) {
+            $source = 'namespace ' . $namespace . ';' . PHP_EOL . $source;
+        }
+
+        // We need to remove following part of code or PHP will crash!
```

Comment:
This is because of the way VCR overwrites generated code by its own code. You can have a look at the generated code, you will see that the use and namespace statements are necessary when the server class references the classmap of the client class. If I duplicate the classmap, this piece of code will probably be easier or unnecessary.